lib/components/AlbumScreen/album_screen_content.dart

```diff
@@ -147,15 +152,15 @@ class _SongsSliverListState extends State<SongsSliverList> {
 
   @override
   Widget build(BuildContext context) {
-    // When user selects song from disc other than first, index number is
-    // incorrect and song with the same index on first disc is played instead.
-    // Adding this offset ensures playback starts for nth song on correct disc.
-    final int indexOffset =
-        widget.childrenForQueue.indexOf(widget.childrenForList[0]);
-
     return SliverList(
       delegate: SliverChildBuilderDelegate(
         (BuildContext context, int index) {
+          // When user selects song from disc other than first, index number is
+          // incorrect and song with the same index on first disc is played instead.
+          // Adding this offset ensures playback starts for nth song on correct disc.
+          final indexOffset =
+              widget.childrenForQueue.indexOf(widget.childrenForList[0]);
+
```

Comment:
I get this exception

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following RangeError was thrown building SongsSliverList(dirty, state:
_SongsSliverListState#a6360):
RangeError (index): Invalid value: Valid value range is empty: 0
```

I once understood why moving it works but I don't remember, it's some special case because we only have 5 displayed songs instead of all of them

Reply:
Try changing it to `final int indexOffset = widget.childrenForList.isNotEmpty ? widget.childrenForQueue.indexOf(widget.childrenForList[0]) : 0` to avoid the exception. If that doesn't work you can move it back into the delegate :)

Reply:
Not needed anymore since everything was moved to its own classes anyways